framework/src/main/java/org/checkerframework/framework/util/dependenttypes/DependentTypesHelper.java

```diff
@@ -600,8 +684,18 @@ private AnnotationMirror standardizeDependentTypeAnnotation(
                     AnnotationUtils.getElementValueArray(anno, value, String.class, true);
             List<String> standardizedStrings = new ArrayList<>();
             for (String expression : expressionStrings) {
-                standardizedStrings.add(
-                        standardizeString(expression, context, localScope, useLocalScope));
+                String standardized =
+                        standardizeString(
+                                expression,
+                                context,
+                                localScope,
+                                useLocalScope,
+                                removeErroneousExpressions);
+                if (standardized == null) {
```

Comment:
Instead of passing `removeErroneousExpressions` to `standardizeString`, this if statement could check if `standardized` is an error string via `DependentTypesError#isExpressionError`.